### PR TITLE
Replace no-method-argument with no-self-argument when variadic args present

### LINKED ISSUE #7507: Method with variadic args/kwargs but no `self` reports `no-method-argument` instead of `no-self-argument`
### Bug description

```py
# pylint: disable=missing-docstring

class MyClass:
    # correct warning: no-method-argument
    def no_args():
        """A method without a self argument or any other arguments."""

    # correct warning: no-self-argument
    def no_self_arg(arg):
        """A method without a self argument but with a standard argument."""

    # incorrect warning: no-method-argument
    def no_self_arg_but_varargs(*args):
        """A method without a self argument but with *args."""

    # incorrect warning: no-method-argument
    def no_self_arg_but_kwargs(**kwargs):
        """A method without a self argument but with **kwargs."""

    # incorrect warning: no-method-argument
    def no_self_arg_but_varargs_and_kwargs(*args, **kwargs):
        """A method without a self argument but with *args and **kwargs."""

    def correct(self):
        """Correct."""
        self.var = "correct"
```

### Configuration

_No response_

### Command used

```shell
pylint repro.py
```

### Pylint output

```shell
************* Module repro
repro.py:5:4: E0211: Method has no argument (no-method-argument)
repro.py:9:4: E0213: Method should have "self" as first argument (no-self-argument)
repro.py:13:4: E0211: Method has no argument (no-method-argument)
repro.py:17:4: E0211: Method has no argument (no-method-argument)
repro.py:21:4: E0211: Method has no argument (no-method-argument)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

I expected to see `no-self-argument` when a method has `*args` or `**kwargs` rather than `no-method-argument` as the latter is quite confusing as the method does have arguments.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.8.14 (default, Sep  8 2022, 00:02:07) 
[GCC 11.2.0]
```

### OS / Environment

Kubuntu 22.04.1 LTS

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jack1142**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Report `no-self-argument` rather than `no-method-argument` for methods with variadic arguments.
Per the comment in #7507, I also added method name to the error messages of no-method-argument and no-self-argument

Closes #7507

### CODE REVIEW

File: doc/whatsnew/fragments/7507.bugfix [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Report `no-self-argument` rather than `no-method-argument` for methods
```
Comment (clavedeluna) [inline doc/whatsnew/fragments/7507.bugfix:1]: You'll want double tick marks, ``...``` for both no-self-arg and no-method-arg so they show up as code

File: doc/whatsnew/fragments/7507.other [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Add method name to the error messages of no-method-argument and no-self-argument.
```
Comment (clavedeluna) [inline doc/whatsnew/fragments/7507.other:1]: may want to add the double tick marks ``..`` here for the msg names

File: pylint/checkers/classes/class_checker.py [hunk 3]
```diff
@@ -1950,8 +1950,13 @@ def _check_first_arg_for_type(
             # Check if there is a decorator which is not named `staticmethod` but is assigned to one.
             return
         # class / regular method with no args
-        elif not node.args.args and not node.args.posonlyargs:
-            self.add_message("no-method-argument", node=node)
+        elif (
+            not node.args.args
+            and not node.args.posonlyargs
+            and not node.args.vararg
+            and not node.args.kwarg
```
Comment (clavedeluna) [inline pylint/checkers/classes/class_checker.py:1957]: could this be refactored to remove so many `not` to
```(
not (node.args.args and node.args.pos...... and ... and )
```

maybe not? worth checking

Comment (jack1142) [reply pylint/checkers/classes/class_checker.py]: Yeah, that makes sense, thanks.

**Review by clavedeluna — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM thank you. Better message in general, and a nice fix to a nonsensical message 👌

**Pull request merged**